CMP1903M Object Oriented Programming A01 2223/CMP1903M A01 2223/CMP1903M A01 2223/Pack.cs

```diff
@@ -8,26 +9,89 @@ namespace CMP1903M_A01_2223
 {
     class Pack
     {
-        List<Card> pack;
+        
+        //Get, set values
+        public int typeOfShuffle { get; set; }
+        public int amount { get; set; }
+
+        //Initialise the card pack here
+        public static List<Card> _pack = new List<Card>();
+        public Card[] cardPack = _pack.ToArray();
 
         public Pack()
         {
-            //Initialise the card pack here
         }
-
-        public static bool shuffleCardPack(int typeOfShuffle)
+        public static bool shuffleCardPack(int typeOfShuffle, int[] cardPack)
         {
             //Shuffles the pack based on the type of shuffle
-
+            if(typeOfShuffle == 1)
+            {
+                Console.WriteLine("Fisher–Yates shuffle");
+                Random _random = new Random(); 
+                int n = cardPack.Length; 
+                //Will change all positions of cards with a random card
+                for(int i = 0; i < n; i++)
+                {
+                    int r = i + _random.Next(n - 1);
+                    int temp = cardPack[r];
+                    cardPack[r] = cardPack[i];
+                    cardPack[i] = temp;
+                }
+                return true;
+            }
+            else if (typeOfShuffle == 2) 
+            {
+                Console.WriteLine("Tabled Card Riffle Shuffle");
+                int n = cardPack.Length;
+                //Creates two halves of the deck 
+                int[] left = { };
+                int[] right = { };
+                int[] shuffledCards = { };
+                //Shuffles with both arrays from both ends
+                for (int i = 0; i < (n / 2); i++) 
+                {
+                    cardPack[i] = left[i];
+                    cardPack[1 - i] = right[i];
+                }
+                for(int i = 0; i < n; i++)
+                {
+                    shuffledCards[i] = left[i];
+                    shuffledCards[i + 1] = right[i];
+                    i++;
+                }
+                return true;
+            }
+            else
+            {
+                //Error handling
+                Console.WriteLine("Invalid number input");
+                return false;
+            }
         }
-        public static Card deal()
+
+        public static Card deal(int[] cardPack)

```

Comment:
Instead of passing cardPack as a parameter you should use the initialized cardPack inside this class. And it should be Card[] not int[]. Because pack holds Card objects not Integers